server/internal/registry/server.go

```diff
@@ -246,69 +275,58 @@ func (s *Local) handlePull(w http.ResponseWriter, r *http.Request) error {
 	defer maybeFlush()
 
 	var mu sync.Mutex
-	enc := json.NewEncoder(w)
-	enc.Encode(progressUpdateJSON{Status: "pulling manifest"})
+	progress := make(map[*ollama.Layer]int64)
 
+	t := time.NewTicker(time.Hour) // "unstarted" timer
+	start := sync.OnceFunc(func() { t.Reset(100 * time.Millisecond) })
 	ctx := ollama.WithTrace(r.Context(), &ollama.Trace{
 		Update: func(l *ollama.Layer, n int64, err error) {
 			mu.Lock()
-			defer mu.Unlock()
-
-			// TODO(bmizerany): coalesce these updates; writing per
-			// update is expensive
-			enc.Encode(progressUpdateJSON{
-				Digest:    l.Digest,
-				Status:    "pulling",
-				Total:     l.Size,
-				Completed: n,
-			})
+			progress[l] = n
+			mu.Unlock()
+			if n > 0 {
+				start()
+			}
 		},
 	})
 
+	pushUpdate := func() {
+		p := progressUpdateJSON{Digest: p.model()}
```

Comment:
Should we use a new field for this now that it's sending back the model name (which I think is a huge positive in the design)

Reply:
I've reverted it back to _mostly_ like it was before. I'll bring back "single" bar later. Trying to fix the CLI in this PR was too ambitious.